### PR TITLE
Add X-XSS-Protection header to nginx config

### PR DESCRIPTION
**Pull request by chmreid**
This PR adds an X-XSS-Protection header to the nginx configuration. This helps avoid cross-site scripting attacks on Dockstore.

Note: this PR also includes changes made by PR #140 and #141 (those changes were included to avoid a merge conflict).

### CODE REVIEW

File: templates/default.nginx_http.conf.template [hunk 1]
```diff
@@ -15,6 +15,18 @@ log_format  custom  '$http_x_forwarded_for - $remote_user [$time_local] "$reques
 # workaround for the way we're including conf https://serverfault.com/questions/729128/overriding-nginx-access-log-directive-duplicate-log-entries
 access_log  off;
 
+# Don't send nginx version number
+server_tokens off;
+
+# Don't allow page to be rendered inside a frame or iframe
```
Comment (denis-yuen) [inline templates/default.nginx_http.conf.template:21]: no iframes? boo

**Comment by david4096** [comment]
Seems like all 3 changes ended up in this PR. I wonder as looking at these whether we will need to add them at the cloudfront layer or if they are passed?

**Comment by chmreid** [comment]
> I wonder as looking at these whether we will need to add them at the cloudfront layer or if they are passed?

The next time dev is updated with the new nginx files, I will check the headers of the site and see if they're showing up (assuming dev is also behind a Cloudfront layer). There may be some configuration of Cloudfront so it will properly forward headers (this document in the Cloudfront documentation is also useful: [caching content based on request headers](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/header-caching.html)).

**Comment by coverbeck** [comment]
> The next time dev is updated with the new nginx files, I will check the headers of the site and see if they're showing up (assuming dev is also behind a Cloudfront layer).

No Dockstore environment is currently behind CloudFront. We do use CloudFront as a way to get assets to the browser more quickly, but the whole site is not behind CloudFront. We had to go with this hybrid approach for now for a couple of reasons; I can point you to some slides if interested. And we hope to get behind CF one day, but we are not there yet.

Our index.html is served through the nginx server. The contents of the index.html reference CF assets, e.g.,

```
  <link rel="stylesheet" href="https://gui.dockstore.org/develop-aec853f/styles.dd63253167824f1d4548.css"></head>
```
So the index.html only is loaded via nginx, but all assets are loaded from CF.

**Comment by chmreid** [comment]
Given that we are including external scripts from Cloudfront, this PR should not be merged yet because it will block external resources from being loaded. First we need a PR to add the `Content-Security-Policy` headers, which can explicitly allow loading external resources (in our case, from Cloudfront URLs).

Related Seabright ticket is SEAB-261

**Comment by chmreid** [comment]
The actual content of this PR was also included with PR #146, which was merged first, so the changes made by this PR became null. The contents of this PR have been merged into the `develop` branch. Closing this PR in favor of #146.

**Pull request closed without merging**